src/components/common/CodeReviewTools.js

```diff
@@ -0,0 +1,401 @@
+import CommitInfoDialog from './CommitInfoDialog';
+
+const locales = { fr, en: enGB };
+
+const styles = (theme) => ({
+  container: {
+    borderSpacing: 0,
+    maxWidth: '800px',
+    width: '90%',
+    margin: '20px auto auto',
+    paddingLeft: '20px',
+    boxSizing: 'border-box',
+    paddingRight: '0px',
+  },
+  toolbar: {
+    paddingRight: '0px !important',
+  },
+  toggleButton: {
+    color: theme.palette.primary.main,
+  },
+  editButton: {
+    color: theme.palette.primary.main,
+  },
+  selectUsers: {
+    width: '100%',
+    display: 'block',
+  },
+  selectVersions: {
+    width: '100%',
+    display: 'block',
+  },
+  infoButton: {
+    color: theme.palette.primary.main,
+  },
+});
+
+const customSelectStyles = (height = '36px') => ({
+  control: (provided) => ({
+    ...provided,
+    minHeight: height,
+    height,
+    maxHeight: height,
+  }),
+});
+
+class CodeReviewTools extends React.Component {
+  static propTypes = {
+    classes: PropTypes.shape({
+      container: PropTypes.string.isRequired,
+      gridRow: PropTypes.string.isRequired,
+      toolbar: PropTypes.string.isRequired,
+      lastGrid: PropTypes.string.isRequired,
+      selectUsers: PropTypes.string.isRequired,
+      selectVersions: PropTypes.string.isRequired,
+      toggleButton: PropTypes.string.isRequired,
+      editButton: PropTypes.string.isRequired,
+      infoButton: PropTypes.string.isRequired,
+    }).isRequired,
+    t: PropTypes.func.isRequired,
+    dispatchOpenEditorView: PropTypes.func.isRequired,
+    dispatchSetCodeEditorSettings: PropTypes.func.isRequired,
+    dispatchOpenCommitInfoDialog: PropTypes.func.isRequired,
+    showVisibilityButton: PropTypes.bool,
+    showEditButton: PropTypes.bool,
+    showHistoryDropdown: PropTypes.bool,
+    hideCommentsCallback: PropTypes.func.isRequired,
+    codeEditorSettings: PropTypes.shape({
+      codeId: PropTypes.string.isRequired,
+      code: PropTypes.string.isRequired,
+    }).isRequired,
+    topBarVisible: PropTypes.bool.isRequired,
+    allVisibleState: PropTypes.bool.isRequired,
+    codeVersions: PropTypes.arrayOf(
+      PropTypes.arrayOf(
+        PropTypes.shape({
+          label: PropTypes.string.isRequired,
+          value: PropTypes.shape({
+            code: PropTypes.string.isRequired,
+            codeId: PropTypes.string.isRequired,
+          }).isRequired,
+        }),
+      ),
+    ).isRequired,
+    codeContributors: PropTypes.arrayOf(
+      PropTypes.shape({
+        label: PropTypes.string.isRequired,
+        value: PropTypes.string.isRequired,
+      }),
+    ).isRequired,
+  };
+
+  static defaultProps = {
+    showVisibilityButton: true,
+    showEditButton: true,
+    showHistoryDropdown: true,
+  };
+
+  state = (() => {
+    const { codeContributors, codeVersions } = this.props;
+    return {
+      selectedBranch: codeContributors[0],
+      selectedVersion: codeVersions[0][0],
+      availableCodeVersions: codeVersions[0],
```

Comment:
Maybe explain with a comment why the `0` indexes.

Reply:
I changed the `0` to constants with explicit names